lib/fluent/plugin/in_s3.rb

```diff
@@ -204,7 +206,13 @@ def run
             body = Yajl.load(message.body)
             log.debug(body)
             next unless body["Records"] # skip test queue
-
+            if @match_regexp
+              s3 = body["Records"].first["s3"]
+              raw_key = s3["object"]["key"]
+              key = CGI.unescape(raw_key)
+              match_regexp = Regexp.new(@match_regexp)
```

Comment:
`@match_regexp` is already `Regexp` object, you can use it directly so that compilation cost can be reduced.

Reply:
Right! 👍 Updated